cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/plugin/MacroParser.java

```diff
@@ -110,11 +120,14 @@ private MacroMetadata findRightmostMacro(String str)
   }
 
   private MacroMetadata getMacroFunctionMetadata(int startIndex, int endIndex, String macroStr, int argsStartIndex) {
-    // check for closing ")" and allow escaping "\)" TODO: and doubly-escaping "\\)"
+    // check for closing ")" and allow escaping "\)" and doubly-escaping "\\)"
     int closingParenIndex = getClosingParenIndex(macroStr);
     if (closingParenIndex < 0 || !macroStr.endsWith(")")) {
       throw new InvalidMacroException(String.format("Could not find enclosing ')' for macro arguments in '%s'.",
                                                     macroStr));
+    } else if (closingParenIndex != macroStr.length() - 1 && !doublyEscaped(closingParenIndex - 1, macroStr)) {
```

Comment:
I don't understand what this condition is for. Won't the `macroStr.endsWith(")")` always trigger before this?